Backend/ServerModules/ServerBehavior.cs

```diff
@@ -31,16 +31,36 @@ protected override void OnMessage(MessageEventArgs e)
             switch(Message.GetMessageType(rawString))
             {
                 case MessageType.TextMessage:
-                    var textMessage = new TextMessage(socket, rawString);
+                    if (user.IsMuted)
+                    {
+                        Send("DO REFUSE\r\nWITH\r\nYou are muted!");
+                        Console.WriteLine($"Failed send attempt from muted user: {user.Username}");
+                        return;
+                    }
+
+                    var textMessage = new TextMessage(currentSocket, rawString);
+                    Console.WriteLine($"{textMessage.Sender}: {textMessage.Content}");
+
+                    foreach (User client in UserManager.UsersList)
```

Comment:
better to wrap this to a function called SendToAll or something. Also do not ignore the current socket because the message needs to be ponged back, check this ticket https://trello.com/c/cZmyfA9M/85-wait-for-server-reflection-on-sent-messages

Reply:
Done